### ISSUE TITLE
Socket cleanout misbehave

### ISSUE THREAD

**Issue by dmitrykruglov** [mongrel2/mongrel2#154]
Mongrel2 should handle socket cleanout in accordance with values of the following settings:
`limits.min_ping` - Minimum time since last activity before considering closing a socket. Defaults to 120. Set to 0 to disable it. 
`limits.min_write_rate` - Minimum bytes/second written before considering closing a socket. Defaults to 300. Set to 0 to disable it. 
`limits.min_read_rate` - Minimum bytes/second read before considering closing a socket. Defaults to 300. Set to 0 to disable it.
`limits.kill_limit` - How many of min_ping, min_write_rate, and min_read_rate have to trigger before a socket is killed. Defaults to 2.

The issue is that if `limits.kill_limit` is set to e.g. 1, a socket will be disconnected if at least 2 of 3 checks trigger, and not 1 as expected. And if it's set to 2 (default value) it will require all 3 checks to trigger in order to disconnect a socket.

This misbehavior is caused by the mistake in this comparison https://github.com/zedshaw/mongrel2/blob/develop/src/register.c#L328
which should be
`if(should_kill >= kill_limit) {`

**Comment by jasom** [comment]
@dimovnike if you set limits.kill_limit to 0, then you should see min_ping kill things.  The logic is that there are 3 limits used to decide when to kill, and if that number is greater than the kill_limit then it will be killed.

**Comment by dimovnike** [comment]
yes, limits.kill_limit=0 works but min_ping is still unreliable. I compiled it with debug info (btw this is broken but I patched makefile to make it compile, there are broken tests and filters that do not support debug). It seems it takes many attempts to kill a socket.  This basically results in min_ping being multiplied 2,3,4,etc times. 

> Mon, 23 Apr 2018 18:49:33 GMT DEBUG src/register.c:307: Checking fd=31:conn_id=846 against last_ping: 20, read_rate: 85515, write_rate: 37893
> Mon, 23 Apr 2018 18:49:33 GMT DEBUG src/register.c:312: Connection fd=31:conn_id=846 over limits.min_ping time: 10 < 20
> Mon, 23 Apr 2018 18:49:33 GMT DEBUG src/register.c:140: Failed to close IOBuffer, probably SSL error.
> Mon, 23 Apr 2018 18:49:33 GMT [WARN] (src/register.c:336: errno: Resource temporarily unavailable) Killed 11 connections according to min_ping: 10, min_write_rate: 0, min_read_rate: 0
> 

NOTE: I do not use SSL yet.

This problem is present in last release and in develop branch.
